main.cpp

```diff
@@ -148,7 +109,19 @@ template <typename NumericType>
 struct Numeric
 {
     using Type = Temporary<NumericType>;
-    explicit Numeric(Type v) : value(std::make_unique<Type>(v)) {}
+
+    Numeric(const NumericType& v) : value(std::make_unique<Type>(v)) {}
+
+    Numeric(NumericType&& v) : value(std::make_unique<Type>(std::forward<NumericType>(v))) {}
```

Comment:
This constructor does not need to exist. 
This is not one of the Rule-Of-5 constructors. 
- [ ] remove this constructor